Nested jars tests

ilbinek opened the pull request.

judovana commented:
Why is the nbin.zip added, removed and again added?

Can the zipbe created in runtime pls? In pre-class?

ilbinek commented:
> Why is the nbin.zip added, removed and again added?
> 
> Can the zipbe created in runtime pls? In pre-class?

Do you mean the .zip file used for the tests?

AurumTheEnd commented:
> > Why is the nbin.zip added, removed and again added?
> > Can the zipbe created in runtime pls? In pre-class?
> 
> Do you mean the .zip file used for the tests?

Yes, the tester.zip file is weirdly added in 8572ac96, removed in e721a7ce, and then re-added in 460278da.

If it is a crucial file for the tests (which I presume it is) and wish to keep it, please rebase it so that the file is added only once.

Actually, looking deeper into it, I presume that 460278da requires the tester.zip file to have different contents than before. In that case, please have the tester.zip file change in 460278da and remove the unnecessary commit e721a7c.

judovana commented:
..or generate it on the fly in pre-class...

ilbinek commented:
> > > Why is the nbin.zip added, removed and again added?
> > > Can the zipbe created in runtime pls? In pre-class?
> > 
> > 
> > Do you mean the .zip file used for the tests?
> 
> Yes, the tester.zip file is weirdly added in [8572ac9](https://github.com/pmikova/java-runtime-decompiler/commit/8572ac96b71e96f9bda398553721134f2b03ef23), removed in [e721a7c](https://github.com/pmikova/java-runtime-decompiler/commit/e721a7ce3117b1999f43707644885074e79a6967), and then re-added in [460278d](https://github.com/pmikova/java-runtime-decompiler/commit/460278da66438be5f7e412e80810f11101deefab).
> 
> If it is a crucial file for the tests (which I presume it is) and wish to keep it, please rebase it so that the file is added only once.
> 
> Actually, looking deeper into it, I presume that [460278d](https://github.com/pmikova/java-runtime-decompiler/commit/460278da66438be5f7e412e80810f11101deefab) requires the tester.zip file to have different contents than before. In that case, please have the tester.zip file change in [460278d](https://github.com/pmikova/java-runtime-decompiler/commit/460278da66438be5f7e412e80810f11101deefab) and drop the unnecessary commit [e721a7c](https://github.com/pmikova/java-runtime-decompiler/commit/e721a7ce3117b1999f43707644885074e79a6967).


8572ac9 Added the zip file.

e721a7c Removed one file in the zip that was not supposed to be inside and is used for testing packing.

460278d Renamed the classes inside the zip so it better represents where they are, as discussed previously.

e721a7c  and 460278d are only changing a small amount of bytes (-169 and +54).

AurumTheEnd commented:
> [8572ac9](https://github.com/pmikova/java-runtime-decompiler/commit/8572ac96b71e96f9bda398553721134f2b03ef23) Added the zip file.
> 
> [e721a7c](https://github.com/pmikova/java-runtime-decompiler/commit/e721a7ce3117b1999f43707644885074e79a6967) Removed one file in the zip that was not supposed to be inside and is used for testing packing.
> 

Ah - misleading commit message. Both Jirka and I interpreted "Removed testing file" as if you had removed the whole zip file, and unfortunately GitHub's UI did not help with distinguishing that you had just "Removed one leftover file from testing archive".

In any case, you are fixing something you yourself introduced in this PR, which has no reason for being a separate commit. Therefore please rebase is so that 8572ac9 already has the correct zip archive, and force-push.

judovana commented:
and generate that zip in runtime...

AurumTheEnd commented:
> and generate that zip in runtime...

Agreed.

ilbinek commented:
> > and generate that zip in runtime...
> 
> Agreed.

Ok, will do

AurumTheEnd commented:
Just as a side-note, did you have to take extra steps in order to manually stage and commit the tester.zip archive? Because our `.gitignore` [clearly forbids](https://github.com/pmikova/java-runtime-decompiler/blob/master/.gitignore#L26) zips. Curious.

ilbinek commented:
Closing PR, will open new one with rewritten implementation.
